chartlets.js/packages/lib/src/plugins/mui/Tabs.tsx

```diff
@@ -41,19 +44,40 @@ export function Tabs({
     }
   };
   return (
-    <MuiTabs id={id} style={style} value={value} onChange={handleChange}>
-      {tabItems?.map((tab) => {
+    <MuiBox sx={{ width: "100%" }}>
+      <MuiBox sx={{ borderBottom: 1, borderColor: "divider" }}>
+        <MuiTabs id={id} style={style} value={value} onChange={handleChange}>
+          {tabItems?.map((tab, index) => {
+            const tabState = isComponentState(tab)
+              ? (tab as TabState)
+              : undefined;
+            return (
+              <MuiTab
+                key={index}
+                label={tabState ? tabState.label : isString(tab) ? tab : ""}
+                icon={
+                  tabState &&
+                  tabState.icon && <MuiIcon>{tabState.icon}</MuiIcon>
+                }
+                disabled={disabled || (tabState && tabState.disabled)}
+              />
+            );
```

Comment:
You can make this cleaner by constructing separate `MuiTab` components, one for the case where `!tabState` and the other where you have it. 

Even better would be to ensure / assert upfront whether `tabItems` contains only `TabState` instances.